src/gromacs/nblib/molecules.h

```diff
@@ -116,6 +114,29 @@ class Molecule
     // returns a sorted vector containing no duplicates of atoms to exclude by indices
     std::vector<std::tuple<int, int>> getExclusions() const;
 
+    //! Get index of the atom with the given unique string identifier
+    size_t getAtomIndex(AtomName atomName);
+
+    //! Add interactions between atoms
+    template <class InteractionType>
+    void addInteraction(InteractionType  interactionType,
```

Comment:
In a Molecule as well, I might suggest that we have one container for each interaction type. This means there is more to manage under the hood, but does non add any complication to how the user interacts with interactions. There is a small potential added bonus that then no lookup is needed and interaction calculations can be dispatched at will once the gromacs interface for listed forces matures.

Reply:
As far as user interaction is concerned, everything is encapsulated within a function called `addInteractions`

Reply:
> As far as user interaction is concerned, everything is encapsulated within a function called addInteractions

Yes, and as far as the implementation is concerned, everything is in one single container which was the point of Joe's comment. I'm also in favor of not using Type Erasure and keep separate containers.